File: leetcode/easy/3483. Unique 3-Digit Even Numbers/main.py
from typing import List, Dict


class Solution:
    def totalNumbers(self, digits: List[int]) -> int:
        seen: Dict[int, bool] = {}
        n = len(digits)

        for i in range(n):
            if digits[i] == 0:
                continue

            for j in range(n):
                if i == j:
                    continue

                for k in range(n):
                    if (k == i) or (k == j) or (digits[k] % 2 != 0):
                        continue

                    seen[digits[i] * 100 + digits[j] * 10 + digits[k]] = True

        return len(seen)

File: leetcode/easy/3483. Unique 3-Digit Even Numbers/main.go
package main

func totalNumbers(digits []int) int {
	seen := make(map[int]struct{})
	n := len(digits)

	for i := range n {
		if digits[i] == 0 {
			continue
		}

		for j := range n {
			if j == i {
				continue
			}

			for k := range n {
				if k == i || k == j || digits[k] % 2 != 0 {
					continue
				}

				seen[digits[i]*100+digits[j]*10+digits[k]] = struct{}{}
			}
		}

	}

	return len(seen)
}
